python/ngen_cal/src/ngen/cal/calibration_set.py

```diff
@@ -19,14 +25,16 @@ class CalibrationSet(Evaluatable):
         A HY_Features based catchment with additional calibration information/functionality
     """
 
-    def __init__(self, adjustables: Sequence[Adjustable], eval_nexus: Nexus, routing_output: 'Path', start_time: str, end_time: str, eval_params: 'EvaluationOptions'):
+    def __init__(self, adjustables: Sequence[Adjustable], eval_nexus: Nexus, hooks: ModelHooks, start_time: str, end_time: str, eval_params: 'EvaluationOptions'):
         """
 
         """
         super().__init__(eval_params)
         self._eval_nexus = eval_nexus
         self._adjustables = adjustables
-        self._output_file = routing_output
+        # record the hooks needed for output and checkpointing
+        self._output_hook = hooks.ngen_cal_model_output
```

Comment:
Just to clean this up a bit, why not just set:

```python
        self._hooks: ModelHooks = hooks
```
and use that where it is needed. Just a little easier to follow. I for sure would be wondering why there was an added layer of indirection if I were to stumble on this code.

Reply:
fair point, changed in 308330ad0a01be638b7757d6eae3f064fbf2f441